### PR TITLE
Clarify and fix grammar for Formats docs

### PR DESCRIPTION
**Pull request by weedySeaDragon**
Like I did with PR #863 , I went through the docs in the **Formats** section and corrected grammar, fixed typos, and did a little clarifying and strengthening.  Like #863, I didn't squash the commits in case someone wanted to review them.  (But I can squash.)

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @weedySeaDragon,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by muuki88** [comment]
:open_mouth:  Wow. Thanks a lot. How long did it take to fix my english? Did you use a tool or was it plain old reading and fixing?

Squashing is not necessary. Github now supports this in the UI ;)

**Comment by weedySeaDragon** [comment]
Just plain old (slightly compulsive) editing.  Writing the comments for the commits is what took a while. ;-)  It wasn't that much, really.
Plus I had a chunk of time to work on it.

There's some really great info in there!  I'm trying to note some ideas for doing a little re-organizing to highlight some of it. (For example:  There's a great explanation of mappings in there.  I think it'd be powerful & much more helpful if that explanation was in the Intro: Main Concepts section.)  That's for another time.  

I'm happy to help.  It's a great way for me to learn this!

File: src/sphinx/formats/debian.rst [hunk 1]
```diff
@@ -95,21 +101,22 @@ Solutions include:
...
 
-and this to your ``plugins.sbt``
+and this to your ``plugins.sbt``:
 
 .. code-block:: scala
 
   libraryDependencies += "org.vafer" % "jdeb" % "1.3" artifacts (Artifact("jdeb", "jar", "jar"))
 
-JDeb is a provided dependency so you have to add it on your own. It brings a lot of dependencies
+JDeb is a provided dependency. (It is not a not an AutoPlugin_.) You have to explicitly add it on your own. It brings a lot of dependencies
```
Comment (muuki88) [inline src/sphinx/formats/debian.rst]: "It is not a not an AutoPlugin._" ? I think you can remove that :)

File: src/sphinx/formats/debian.rst [hunk 2]
```diff
@@ -210,7 +220,10 @@ The helper methods can be found in `MaintainerScriptHelper Scaladocs`_.
 
 If you use the ``JavaServerAppPackaging`` there are predefined ``postinst`` and
 ``preinst`` files, which start/stop the application on install/remove calls. Existing
-maintainer scripts will be extended not overridden.
+maintainer scripts will be *extended* not overridden.
+
+Use a Different Castle Directory for your Control Scripts
```
Comment (muuki88) [inline src/sphinx/formats/debian.rst]: Ah :( I guess you can remove the bad jokes here as well ;)

File: src/sphinx/formats/docker.rst [hunk 3]
```diff
@@ -55,9 +56,12 @@ and this to your ``plugins.sbt``
 
   libraryDependencies += "com.spotify" % "docker-client" % "3.5.13"
 
-The Docker-spotify client is a provided dependency so you have to add it on your own.
-It brings a lot of dependenciesthat could slow your build times. This is the reason
-the dependency is marked as provided.
+The Docker-spotify client is a provided dependency. (It is not a not an AutoPlugin_.) You have to explicitly add it on your own. It brings a lot of dependencies
```
Comment (muuki88) [inline src/sphinx/formats/docker.rst]: Again "It is not a not an AutoPlugin._" ?

File: src/sphinx/formats/docker.rst [hunk 4]
```diff
@@ -85,7 +89,7 @@ Informational Settings
     The version of the package for Docker (if different from general version).  Often takes the form ``x.y.z``.
 
   ``maintainer in Docker``
-    The maintainer of the package, required by the Dockerfile format.
+    The maintainer of the package, required by the Docker file format.
```
Comment (muuki88) [inline src/sphinx/formats/docker.rst]: I think `Dockerfile` is the correct name here. See https://docs.docker.com/engine/reference/builder/

File: src/sphinx/formats/universal.rst [hunk 5]
```diff
@@ -351,46 +360,91 @@ You get a set of methods which will help you to create mappings very easily.
...
+
+    mappings in Universal ++= directory("SomeDirectoryNameToInclude")
+
+If you want to add everything in a directory where the path for the directory is dynamic, e.g. the ``scala-2.10/api`` directory that is nested under in the ``target`` directory, and ``target`` is defined in a task:
+
+.. code-block:: scala
+
+    mappings in Universal <++= (packageBin in Compile, target) map { (_, target) =>
+      directory(target / "scala-2.10" / "api")
+    }
```
Comment (muuki88) [inline src/sphinx/formats/universal.rst]: This is a nice example. Can you change it, so it doesn't use the `<++=` syntax? We should get rid of this archaic and confusing sbt syntax. So instead:

```scala
mappings in Universal := {
   (mappings in Universal).value ++ directory(target.value / "scala-2.10" / "api")
}
```

File: src/sphinx/formats/universal.rst [hunk 6]
```diff
@@ -351,46 +360,91 @@ You get a set of methods which will help you to create mappings very easily.
...
-      directory(target / "scala-2.10" / "api")
-    }
+    ``dir.getParentFile``  returns the parent of ``dir``.  In this example, it's the parent directory of whatever ``target`` is.
+
+    ``pair`` is a PathFinder_ method that takes a function and applies it to every file (in the sequence), and returns a *(file, function-result)* tuple.
 
-    //For static content it can be added to mappings directly
-    mappings in Universal ++= directory("SomeResourcesToInclude")
+Putting it all together, this creates a map of every file under ``target/scala-2.10/api`` (including the directory ``target/scala-2.10/api`` itself)
+with a string that is the path to the parent of ``target``.  This is a mapping for every file and a string that tells the universal packager where it is located.
```
Comment (muuki88) [inline src/sphinx/formats/universal.rst]: This is really a lot better. Thanks :)

File: src/sphinx/formats/windows.rst [hunk 7]
```diff
@@ -3,12 +3,12 @@
...
 ==============
 
-The windows packaging is completely tied to the WIX installer toolset.  For any non-trivial package,
+Windows packaging is completely tied to the WIX installer toolset.  For any non-trivial package,
 it's important to understand how WIX works.  http://wix.tramontana.co.hu/ is an excellent tutorial
 to how to create packages using wix.
 
 However, the native-packager provides a simple layer on top of wix that *may* be enough for most projects.
-If it is not, just override ``wixConfig`` or ``wixFile`` settings.  Let's look at the layer above direct
+If it is not enought, just override ``wixConfig`` or ``wixFile`` settings.  Let's look at the layer above direct
```
Comment (muuki88) [inline src/sphinx/formats/windows.rst]: enough*t*

**Comment by muuki88** [comment]
So a few small notes. Beside that: Amazing work :)

File: src/sphinx/formats/debian.rst (hunk 1 above)
Comment (weedySeaDragon) [inline src/sphinx/formats/debian.rst]: Will do.  (I think that shows my personal learning curve. :-) )

File: src/sphinx/formats/docker.rst (hunk 3 above)
Comment (weedySeaDragon) [inline src/sphinx/formats/docker.rst]: I'll remove it.

File: src/sphinx/formats/docker.rst (hunk 4 above)
Comment (weedySeaDragon) [inline src/sphinx/formats/docker.rst]: Ya.  I think I let autocorrect go too far.  Will fix.

File: src/sphinx/formats/universal.rst (hunk 5 above)
Comment (weedySeaDragon) [inline src/sphinx/formats/universal.rst]: Will do.  
 I think this is one of the examples that would be great in the intro / main concepts section.  (For a later PR)

Comment (muuki88) [inline src/sphinx/formats/universal.rst]: Really looking forward to see another improvement in the docs :)

**Comment by weedySeaDragon** [comment]
Will make those changes in a bit.  Thanks for reviewing it so quickly!

**Comment by muuki88** [comment]
Awesome :) You made it easy to review ;)

**Pull request merged**